### ISSUE TITLE
Unspecified GSS failure in authenticate_gss_server_step(gss_accept_sec_context)

### ISSUE THREAD

**Issue by gentlebells** [dmansfield/passport-negotiate#6]
Hi
I am trying to authenticate with passport-negotiate in Ubuntu. And KDC server is running on Windows AD and I've got the following error.
I have read several posts and have tried many things, but it was difficult to find the cause of the error.
Is there anything I need to check further?
Thanks,
Jo

Platform: Ubuntu 14.04 LTS
KDC: Windows 2008

<pre>
<code>
Error: authenticate_gss_server_step(gss_accept_sec_context): An unsupported mechanism was requested: Unknown error
   at Error (native)
</code>
</pre>

<pre>
<code>
passport.use('login', new NegotiateStrategy({
        servicePrincipalName: 'HTTP@test-host.testserver.com',
        verbose: true
  }, function (principal, done) {
    console.log('***************');
    console.log(principal);
    console.log(arguments);
    done(null, principal);
  }));
</code>
</pre>

<pre>
<code>
$ klist -k /etc/krb5.keytab
Keytab name: FILE:/etc/krb5.keytab
KVNO Principal
---- --------------------------------------------------------------------------
  22 HTTP/test-host.testserver.com@TESTSERVER.COM

$ ls -al /etc/krb5.keytab
-rwxrwxrwx 1 root root 99 Nov 10 17:08 /etc/krb5.keytab
</code>
</pre>

**Comment by ophers** [comment]
I don't think you need to specify 'servicePrincipalName' and if you do, shouldn't it be identical to the one in your keytab?
